pkg/block/s3/adapter.go

```diff
@@ -93,7 +94,13 @@ func WithTranslator(t block.UploadIDTranslator) func(a *Adapter) {
 	}
 }
 
-func NewAdapter(awsSession *session.Session, opts ...func(a *Adapter)) *Adapter {
+func WithStatsCollector(s stats.Collector) func(a *Adapter) {
+	return func(a *Adapter) {
+		a.clients = a.clients.WithStatsCollector(s)
+	}
+}
+
+func NewAdapter(_ context.Context, awsSession *session.Session, opts ...func(a *Adapter)) *Adapter {
```

Comment:
now we can remove the ctx we added in this pr

Reply:
oops